profile/src/v100/address/account_address.rs

```diff
@@ -0,0 +1,359 @@
+use crate::prelude::*;
+
+/// The address of an Account, a bech32 encoding of a public key hash
+/// that starts with the prefix `"account_"`, dependent on NetworkID, meaning the same
+/// public key used for two AccountAddresses on two different networks will not have
+/// the same address.
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    Eq,
+    Hash,
+    SerializeDisplay,
+    DeserializeFromStr,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[display("{address}")]
+pub struct AccountAddress {
+    /// Human readable address of an account. Always starts with `"account_"``, for example:
+    ///
+    /// `account_rdx16xlfcpp0vf7e3gqnswv8j9k58n6rjccu58vvspmdva22kf3aplease`
+    ///
+    /// Most commonly the user will see this address in its abbreviated
+    /// form which is:
+    ///
+    /// `acco...please`
+    ///
+    /// Addresses are checksummed, as per Bech32. **Only** *Account* addresses starts with
+    /// the prefix `account_`.
+    pub address: String,
+
+    /// The network this account address is tied to, i.e. which was used when a public key
+    /// hash was used to bech32 encode it. This means that two public key hashes will result
+    /// in two different account address on two different networks.
+    pub network_id: NetworkID,
+}
+
+#[uniffi::export]
+pub fn new_account_address(bech32: String) -> Result<AccountAddress> {
```

Comment:
UniFFI does not support methods on `uniffi::Record`, so must declare these as global freestanding functions.